src/sentry/utils/transitional_serializer.py

```diff
@@ -0,0 +1,26 @@
+from json.decoder import JSONDecodeError
+from typing import Any, Dict
+
+from django.contrib.sessions.serializers import PickleSerializer
+from django.core.signing import JSONSerializer
+
+from sentry.utils import metrics
+
+
+class TransitionalSerializer:
+    def __init__(self) -> None:
+        self.pickle_serializer = PickleSerializer()
+        self.json_serializer = JSONSerializer()
+
+    def dumps(self, obj: Dict[str, Any]) -> bytes:
+        # will have the write as pickle obj for now, will change to json in seperate deploy
+        # return self.json_serializer.dumps(obj)
+        return self.pickle_serializer.dumps(obj)
+
+    def loads(self, data: bytes) -> Dict[str, Any]:
+        try:
+            metrics.incr("transitional_serializer_json_read")
```

Comment:
I would have these be `transitional_serializer.json_read` (dot scopes it which may make datadog search a bit better). I would also add one metric for the picked write as well.